node/src/BaseClient.ts

```diff
@@ -2633,6 +2635,36 @@ export class BaseClient {
         return this.createWritePromise(createObjectRefcount(key));
     }
 
+    /**
+     * Returns the index of the first occurrence of `element` inside the list specified by `key`. If no
+     * match is found, `null` is returned. If the `count` option is specified, then the function returns
+     * an `array` of indices of matching elements within a list.
+     *
+     * See https://valkey.io/commands/lpos/ for more details.
+     *
+     * since - Valkey version 6.0.6.
+     *
+     * @param key - The name of the list.
+     * @param element - The value to search for within the list.
+     * @param options - The LPOS options.
+     * @returns The index of `element`, or `null` if `element` is not in the list. If the `count` option
+     * is specified, then the function returns an `array` of indices of matching elements within a list.
+     *
+     * @example
+     * ```typescript
+     * await client.rpush("myList", ["a", "b", "c", "d", "e", "e"]);
+     * console.log(await client.lpos("myList", "e", { rank: 2 })); // Output: 5 - the second occurence of "e" is in the fifth element.
+     * console.log(await client.lpos("myList", "e", { count: 3 })); // Output: [ 4, 5, 6 ] - the elements of the occurences of "e".
+     * ```
+     */
+    public lpos(
+        key: string,
+        element: string,
+        options?: LPosOptions,
```

Comment:
Java has multiple function signatures since it doesn't allow optional args, eg lpos with/without options, lposCount with/without options, thus it also has its own LPosOptions class. Since Node allows optional args like python, I think we should follow what python does: instead of creating an LPosOptions class, all the individual options are provided as optional args (see core.py#lpos).

Reply:
Unfortunately I haven't found a good way to call the function with multiple optional parameters while skipping some of the parameters.